src/generate.js

```diff
@@ -4,60 +4,94 @@ var self = module.exports;
 self.generate = function(options) {
-	// Set defaults.
-	if (!options.hasOwnProperty('length')) options.length = 10;
-	if (!options.hasOwnProperty('numbers')) options.numbers = false;
-	if (!options.hasOwnProperty('symbols')) options.symbols = false;
-	if (!options.hasOwnProperty('uppercase')) options.uppercase = true;
-	if (!options.hasOwnProperty('excludeSimilarCharacters')) options.excludeSimilarCharacters = false;
-
-	var lowercase = 'abcdefghijklmnopqrstuvwxyz',
-		uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
-		numbers = '0123456789',
-		symbols = '!@#$%^&*()+_-=}{[]|:;"/?.><,`~',
-		similarCharacters = /[ilLI|`oO0]/g;
-
-	// Generate character pool
-	var pool = lowercase;
-
-	// uppercase
-	if (options.uppercase) {
-		pool += uppercase;
-	}
-	// numbers
-	if (options.numbers) {
-		pool += numbers;
-	}
-	// symbols
-	if (options.symbols) {
-		pool += symbols;
-	}
-	// similar characters
-	if (options.excludeSimilarCharacters) {
-		pool = pool.replace(similarCharacters, '');
-	}
-
-	var password = '';
-	for (var i = 0; i < options.length; i++) {
-		password += pool[randomNumber(pool.length)];
-	}
-
-	return password;
+  // Set defaults.
+  if (!options.hasOwnProperty('length')) options.length = 10;
+  if (!options.hasOwnProperty('numbers')) options.numbers = false;
+  if (!options.hasOwnProperty('symbols')) options.symbols = false;
+  if (!options.hasOwnProperty('uppercase')) options.uppercase = true;
+  if (!options.hasOwnProperty('excludeSimilarCharacters')) options.excludeSimilarCharacters = false;
+  if (!options.hasOwnProperty('strict')) options.strict = false;
+
+  // Generate character pool
+  var pool = lowercase;
+
+  // uppercase
+  if (options.uppercase) {
+    pool += uppercase;
+  }
+  // numbers
+  if (options.numbers) {
+    pool += numbers;
+  }
+  // symbols
+  if (options.symbols) {
+    pool += symbols;
+  }
+  // similar characters
+  if (options.excludeSimilarCharacters) {
+    pool = pool.replace(similarCharacters, '');
+  }
+
+  var password = '';
+
+  if(options.strict){
+    var minStrictLength = 1 + (options.numbers ? 1 : 0) + (options.symbols ? 1 : 0) + (options.uppercase ? 1 : 0);
+    if(minStrictLength > options.length){
+      throw(new Error('Length should correlate with strict guidelines'));
+    } else {
+      var strictGroup = [];
+
+      strictGroup.push(lowercase[randomNumber(lowercase.length)]);
+
+      if (options.uppercase) {
+        strictGroup.push(uppercase[randomNumber(uppercase.length)]);
+      }
+      // numbers
+      if (options.numbers) {
+        strictGroup.push(numbers[randomNumber(numbers.length)]);
+      }
+      // symbols
+      if (options.symbols) {
+        strictGroup.push(symbols[randomNumber(symbols.length)]);
+      }
+
+      for (var j = minStrictLength; j < options.length; j++) {
+        strictGroup.push(pool[randomNumber(pool.length)]);
+      }
+
+      strictGroup.sort(function(){ return randomNumber(2) - 0.5;});
+      password = strictGroup.join('');
```

Comment:
This **sort** function reorders all of the list.
However there are expected edge cases where if length of the string is 2 symbols then knowing if it's a number is 50%. If we have 3 symbols. Chances that the third symbol is a letter is higher because the pool consist 26 letters and only 10 numbers. 
The think I am going for is that deterministically you can only know that the sequence will have at least one required value. You don't know where, but chance of knowing decrease by the number of letters. So more letters you have the lower the chance of guessing where the correct letter is.
Hope it makes seanse.